### PR TITLE
fix optional storage operator issue

### PR DESCRIPTION
**Pull request by yunjiang29**
Add `ipi-conf-manual-creds-remove-unnecessary-creds` step to following chains:
```
cucushift-installer-rehearse-aws-c2s-ipi-disconnected-private-provision-chain.yaml
cucushift-installer-rehearse-aws-ipi-disconnected-private-cco-manual-security-token-service-provision-chain.yaml
cucushift-installer-rehearse-aws-ipi-disconnected-private-cco-manual-security-token-service-with-ep-sts-ec2-elb-provision-chain.yaml
cucushift-installer-rehearse-aws-ipi-disconnected-private-provision-chain.yaml
cucushift-installer-rehearse-aws-ipi-disconnected-private-sdn-provision-chain.yaml
cucushift-installer-rehearse-aws-ipi-disconnected-private-sno-provision-chain.yaml
cucushift-installer-rehearse-aws-ipi-disconnected-private-techpreview-provision-chain.yaml
cucushift-installer-rehearse-aws-ipi-private-cco-manual-security-token-service-provision-chain.yaml
cucushift-installer-rehearse-aws-ipi-proxy-cco-manual-security-token-service-provision-chain.yaml
cucushift-installer-rehearse-gcp-ipi-cco-manual-workload-identity-provision-chain.yaml
```

### CODE REVIEW

**Comment by yunjiang29** [comment]
@jinyunma @jianlinliu @gpei PTAL

File: ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/disconnected/private/ovn/provision/cucushift-installer-rehearse-gcp-ipi-disconnected-private-ovn-provision-chain.yaml [hunk 1]
```diff
@@ -11,7 +11,8 @@ chain:
   - ref: ipi-conf-telemetry
   - ref: ipi-conf-gcp
   - ref: ipi-conf-mirror
-  - ref: ovn-conf 
+  - ref: ovn-conf
+  - ref: ipi-conf-manual-creds-remove-unnecessary-creds
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/disconnected/private/ovn/provision/cucushift-installer-rehearse-gcp-ipi-disconnected-private-ovn-provision-chain.yaml:15]: I think this workflow does not need this step.

File: ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/disconnected/private/provision/cucushift-installer-rehearse-gcp-ipi-disconnected-private-provision-chain.yaml [hunk 2]
```diff
@@ -11,6 +11,7 @@ chain:
   - ref: ipi-conf-telemetry
   - ref: ipi-conf-gcp
   - ref: ipi-conf-mirror
+  - ref: ipi-conf-manual-creds-remove-unnecessary-creds
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/disconnected/private/provision/cucushift-installer-rehearse-gcp-ipi-disconnected-private-provision-chain.yaml:14]: I think this workflow does not need this step.

File: ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/disconnected/private/techpreview/provision/cucushift-installer-rehearse-aws-ipi-disconnected-private-techpreview-provision-chain.yaml [hunk 3]
```diff
@@ -14,6 +14,7 @@ chain:
   - ref: ipi-install-monitoringpvc
   - ref: ipi-conf-techpreview
   - ref: proxy-config-generate
+  - ref: ipi-conf-manual-creds-remove-unnecessary-creds
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/disconnected/private/techpreview/provision/cucushift-installer-rehearse-aws-ipi-disconnected-private-techpreview-provision-chain.yaml:17]: I think this workflow does not need this step.

**Comment by jianlinliu** [comment]
I guess the following files also need the change:
```
[jianlinliu@jialiu-laptop rehearse]$ grep -r "ipi-conf-manual-creds"
gcp/ipi/cco-manual-workload-identity/provision/cucushift-installer-rehearse-gcp-ipi-cco-manual-workload-identity-provision-chain.yaml:  - ref: ipi-conf-manual-creds
aws/c2s/ipi/disconnected/private/provision/cucushift-installer-rehearse-aws-c2s-ipi-disconnected-private-provision-chain.yaml:    - ref: ipi-conf-manual-creds
aws/ipi/proxy/cco-manual-security-token-service/provision/cucushift-installer-rehearse-aws-ipi-proxy-cco-manual-security-token-service-provision-chain.yaml:  - ref: ipi-conf-manual-creds
aws/ipi/private/cco-manual-security-token-service/provision/cucushift-installer-rehearse-aws-ipi-private-cco-manual-security-token-service-provision-chain.yaml:  - ref: ipi-conf-manual-creds
```

File: ci-operator/step-registry/ipi/conf/manual-creds/remove-unnecessary-creds/ipi-conf-manual-creds-remove-unnecessary-creds-ref.yaml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  as: ipi-conf-manual-creds-remove-unnecessary-creds
+  from_image:
+   namespace: ocp
+   name: "4.12"
+   tag: upi-installer
+  commands: ipi-conf-manual-creds-remove-unnecessary-creds-commands.sh
+  resources:
+    requests:
+      cpu: 10m
+      memory: 100Mi
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/manual-creds/remove-unnecessary-creds/ipi-conf-manual-creds-remove-unnecessary-creds-ref.yaml:11]: I think this step need to  set `BASELINE_CAPABILITY_SET` and `ADDITIONAL_ENABLED_CAPABILITY_SET` to receive the values passed from workflow level.

File: ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/disconnected/private/ovn/provision/cucushift-installer-rehearse-gcp-ipi-disconnected-private-ovn-provision-chain.yaml (hunk 1 above)
Comment (yunjiang29) [reply ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/disconnected/private/ovn/provision/cucushift-installer-rehearse-gcp-ipi-disconnected-private-ovn-provision-chain.yaml]: Updated.

File: ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/disconnected/private/provision/cucushift-installer-rehearse-gcp-ipi-disconnected-private-provision-chain.yaml (hunk 2 above)
Comment (yunjiang29) [reply ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/disconnected/private/provision/cucushift-installer-rehearse-gcp-ipi-disconnected-private-provision-chain.yaml]: Updated.

File: ci-operator/step-registry/ipi/conf/manual-creds/remove-unnecessary-creds/ipi-conf-manual-creds-remove-unnecessary-creds-ref.yaml (hunk 4 above)
Comment (yunjiang29) [reply ci-operator/step-registry/ipi/conf/manual-creds/remove-unnecessary-creds/ipi-conf-manual-creds-remove-unnecessary-creds-ref.yaml:11]: Updated.

File: ci-operator/step-registry/ipi/conf/manual-creds/remove-unnecessary-creds/ipi-conf-manual-creds-remove-unnecessary-creds-commands.sh [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+            [ $? -ne 0 ] && echo "[ERROR] error remove file ${i}" && return 1
+        fi
+    done
+    popd
+    return 0
+}
+
+if (( ocp_minor_version == 12 && ocp_major_version == 4 )); then
+    # 4.12, if Storage operator not installed, then remove CSI driver CR
+    if [[ "${BASELINE_CAPABILITY_SET}" == "None" ]] && [[ ! "${ADDITIONAL_ENABLED_CAPABILITY_SET}" =~ "Storage" ]]; then
```
Comment (jinyunma) [inline ci-operator/step-registry/ipi/conf/manual-creds/remove-unnecessary-creds/ipi-conf-manual-creds-remove-unnecessary-creds-commands.sh:41]: v4.11 set is {baremetal, openshift-samples, marketplace}.
If ${BASELINE_CAPABILITY_SET} set to `v4.11`, no additionalCapabilitySet defined or not including storage co,  storage is also disabled.

**Comment by yunjiang29** [comment]
@jianlinliu @jinyunma @gpei the script works as expected. 
1 (baseCap: None) failed due to monitorvpc issue, and the other 3 failed due to VPC quota.

ready for merge

**Comment by jianlinliu** [comment]
@yunjiang29 something wrong on https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/33641/rehearse-33641-periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-e2e-aws-usgov-ipi-disconnected-private-p2/1588063960370778112/artifacts/e2e-aws-usgov-ipi-disconnected-private-p2/ipi-conf-manual-creds-remove-unnecessary-creds/build-log.txt, the workflow should NOT remove the storage.

**Comment by yunjiang29** [comment]
> the workflow should NOT remove the storage.

yes, I noticed that and fixed

```
*)
  base_operators="${vCurrent}" # include all operators by default
  ;;
```

**Comment by yunjiang29** [comment]
@jinyunma @jianlinliu @gpei PR is ready for merge.
The failure jobs are: 1 mirror images issue, a flake, 1 monitorvpc issue, which is expected, the others are e2e tests or known destroy issue.

/unhold

**Pull request merged**